gradually-update/src/lib.rs

```diff
@@ -0,0 +1,155 @@
+	}
+}
+
+decl_event!(
+	/// Event for gradually-update module.
+	pub enum Event<T> where
+	<T as frame_system::Trait>::BlockNumber,
+	{
+		/// Add gradually_update success (key, per_block, target_value)
+		GraduallyUpdate(StorageKey, StorageValue, StorageValue),
+		/// Cancel gradually_update success (key)
+		CancelGraduallyUpdate(StorageKey),
+		/// Update gradually_update success (blocknum, key, target_value)
+		GraduallyUpdateBlockNumber(BlockNumber, StorageKey, StorageValue),
+	}
+);
+
+decl_error! {
+	/// Error for gradually-update module.
+	pub enum Error for Module<T: Trait> {
+		InvalidTargetValue,
+		GraduallyUpdateHasExisted,
+		CancelGradullyUpdateNotExisted,
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		type Error = Error<T>;
+
+		fn deposit_event() = default;
+
+		const UpdateFrequency: T::BlockNumber = T::UpdateFrequency::get();
+
+		/// Add gradually_update to adjust numeric parameter. This is a root call.
+		pub fn gradually_update(origin, update: GraduallyUpdate) {
+			ensure_root(origin)?;
+
+			// Support max value is u128, ensure per_block and target_value <= 16 bytes.
+			// If current_value is zero, it's length is zero, so can't compare with target_value.len().
+			ensure!(update.per_block.len() == update.target_value.len() && update.per_block.len() <= 16, Error::<T>::InvalidTargetValue);
+
+			let mut gradually_updates = GraduallyUpdates::get();
+			ensure!(!gradually_updates.contains(&update), Error::<T>::GraduallyUpdateHasExisted);
+
+			gradually_updates.push(update.clone());
+			GraduallyUpdates::put(gradually_updates);
+
+			Self::deposit_event(RawEvent::GraduallyUpdate(update.key, update.per_block, update.target_value));
+		}
+
+		/// Cancel gradually_update to adjust numeric parameter. This is a root call.
+		pub fn cancel_gradually_update(origin, key: StorageKey) {
+			ensure_root(origin)?;
+
+			let gradually_updates: Vec<GraduallyUpdate> = GraduallyUpdates::get()
+				.into_iter()
+				.filter(|item| item.key != key)
+				.collect();
+
+			ensure!(GraduallyUpdates::get().len() - gradually_updates.len() == 1, Error::<T>::CancelGradullyUpdateNotExisted);
+			GraduallyUpdates::put(gradually_updates);
+
+			Self::deposit_event(RawEvent::CancelGraduallyUpdate(key));
+		}
+
+		/// Update gradually_update to adjust numeric parameter.
+		fn on_finalize(now: T::BlockNumber) {
+			Self::_on_finalize(now);
+		}
+	}
+}
+
+impl<T: Trait> Module<T> {
+	fn _on_finalize(now: T::BlockNumber) {
+		if now < GraduallyUpdateBlockNumber::<T>::get() + T::UpdateFrequency::get() {
+			return;
+		}
+
+		let mut gradually_updates = GraduallyUpdates::get();
+		for (i, update) in gradually_updates.clone().iter().enumerate() {
+			let current_value = storage::unhashed::get::<StorageValue>(&update.key).unwrap_or_default();
+			let current_value_u128 = u128::from_le_bytes(Self::convert_vec_to_u8(&current_value));
+
+			let frequency_u128 =
```

Comment:
```suggestion
			let frequency_u128: u128 = T::UpdateFrequency::get().saturated_into();
```
I think this should work